File: client/app/types/message.ts
/** Multimodal message body (corresponds to Python type/__init__.py MultiModalMessage) */
export interface MultiModalMessage {
  /** Text content */
  text: string;
  /** Image path list */
  image_path_list?: string[];
  /** Image byte data (base64 string) */
  image_bytes_list?: string[];
  /** Image base64 list */
  image_base64_list?: string[];
  /** Audio path list */
  audio_path_list?: string[];
  /** Audio byte data (base64 string) */
  audio_bytes_list?: string[];
  /** Video path list */
  video_path_list?: string[];
  /** Video byte data (base64 string) */
  video_bytes_list?: string[];
}

File: pub/types/message.py
"""Multimodal message models shared between agent and channels."""

from pydantic import BaseModel


class MultiModalMessage(BaseModel):
    """Multimodal chat message (text plus optional media lists)."""

    # Text content
    text: str

    # Images (multiple supported)
    image_path_list: list[str] | None = None
    image_bytes_list: list[bytes] | None = None
    image_base64_list: list[str] | None = None

    # Audio (single only)
    audio_path_list: list[str] | None = None
    audio_bytes_list: list[bytes] | None = None

    # Video (single only)
    video_path_list: list[str] | None = None
    video_bytes_list: list[bytes] | None = None
